### PR TITLE
Evolution updates

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
Addresses an oversight brought up by PCG inside the Discord server.

## **Discord contact info**
Lunos#4026

### CODE REVIEW

File: include/config/item.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #define I_HEALTH_RECOVERY       GEN_LATEST  // In Gen7+, certain healing items recover a different amount of HP than they used to.
 #define I_SITRUS_BERRY_HEAL     GEN_LATEST  // In Gen4+, Sitrus Berry was changed from healing 30 HP to healing 25% of Max HP.
 #define I_VITAMIN_EV_CAP        GEN_LATEST  // In Gen8, the Vitamins no longer have a cap of 100 EV per stat.
+#define I_RAZOR_CLAW_ON_FIELD   GEN_LATEST  // In PLA (here Gen8+), the Razor Claw can be used as an evolution item on the field.
```
Comment (AsparagusEduardo) [inline include/config/item.h:11]: This shouldn't be a config. We should allow all evolution held items to be usable like stones now that we have official uses, like how we added Linking Cord evolutions.

File: include/constants/pokemon.h [hunk 2]
```diff
@@ -307,6 +307,7 @@
 #define EVO_SCRIPT_TRIGGER_DMG            36     // Pokémon has specified HP below max, then player interacts trigger
 #define EVO_DARK_SCROLL                   37     // interacts with Scroll of Darkness
 #define EVO_WATER_SCROLL                  38     // interacts with Scroll of Waters
+#define EVO_ITEM_DAY                      39     // specified item is used on Pokémon, is day
```
Comment (AsparagusEduardo) [inline include/constants/pokemon.h:310]: I would add `EVO_ITEM_NIGHT` for completeness' sake.

In that same subject, `EVO_ITEM_HOLD` would also be useful for users, as I've seen multiple people ask for it.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: include/config/item.h (hunk 1 above)
Comment (LOuroboros) [reply include/config/item.h:11]: Sounds good to me. I'll get to it.

File: include/constants/pokemon.h (hunk 2 above)
Comment (LOuroboros) [reply include/constants/pokemon.h:310]: Sure, why not.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
It's still missing updating Gligar, Happiny, and regular Sneasel evolutions, not to mention adding Hisuian Sneasel's `EVO_ITEM_HOLD_DAY` method.

**Comment by LOuroboros** [comment]
> It's still missing updating Gligar, Happiny

Will fix.

> and regular Sneasel evolutions

?

> not to mention adding Hisuian Sneasel's `EVO_ITEM_HOLD_DAY` method.

Wot? It's not hold, it's used. And it was like, the thing this PR was originally addressing.

**Comment by LOuroboros** [comment]
Ah, you mean to extrapolate things and make Sneasel be able to evolve into Weavile via `EVO_ITEM_NIGHT`, and for Hisuian Sneasel to be able to evolve into Sneasler via `EVO_ITEM_HOLD_DAY`? Is that it @AsparagusEduardo?

**Comment by AsparagusEduardo** [comment]
> Ah, you mean to extrapolate things and make Sneasel be able to evolve into Weavile via `EVO_ITEM_NIGHT`, and for Hisuian Sneasel to be able to evolve into Sneasler via `EVO_ITEM_HOLD_DAY`? Is that it @AsparagusEduardo?

Ye

**Comment by AsparagusEduardo** [comment]
Please rebase the PR to upcoming.

Also, based on discussion in the Discord server, it might be better to have a config for the items themselves to determine if they can be used for evolution or not, instead of modifying evolution data.

**Comment by LOuroboros** [comment]
> Also, based on discussion in the Discord server, it might be better to have a config for the items themselves to determine if they can be used for evolution or not, instead of modifying evolution data.

Can you elaborate?

You mean a config just like the one I originally added for the Razor Claw, but that encompasses all the items that are being turned into evolution items here, or...?

**Comment by AsparagusEduardo** [comment]
> You mean a config just like the one I originally added for the Razor Claw, but that encompasses all the items that are being turned into evolution items here, or...?

Yes. The evolution data is good as it is, the config would only be for the items themselves.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**